### PR TITLE
Add a job to execute the Eventing tests with the eventingWebhookAuth flag enabled

### PR DESCRIPTION
**Pull request by marcobebway**
**Description**

Add a job to execute the Eventing tests with the eventingWebhookAuth flag enabled.

**Successful runs**

- When the flag is [enabled](https://status.build.kyma-project.io/view/gs/kyma-prow-logs/pr-logs/pull/kyma-project_kyma/17798/marcobebway_test_of_prowjob_pre-main-kyma-gardener-gcp-eventing/1679810661556686848).
- When the flag is [disabled](https://status.build.kyma-project.io/view/gs/kyma-prow-logs/pr-logs/pull/kyma-project_kyma/17798/marcobebway_test_of_prowjob_pre-main-kyma-gardener-gcp-eventing/1679816590243139584).

**Related issue(s)**

- https://github.com/kyma-project/kyma/issues/17797

### CODE REVIEW

**Review by mfaizanse — DISMISSED** [review]

File: prow/jobs/kyma/kyma-integration-gardener-eventing.yaml [hunk 1]
```diff
@@ -148,4 +148,74 @@ presubmits: # runs on PRs
...
+      labels:
+        prow.k8s.io/pubsub.project: "sap-kyma-prow"
+        prow.k8s.io/pubsub.runID: "pre-main-kyma-gardener-gcp-eventing-with-auth-manager"
+        prow.k8s.io/pubsub.topic: "prowjobs"
+        preset-build-pr: "true"
+        preset-cluster-version: "true"
+        preset-debug-commando-oom: "true"
+        preset-gardener-gcp-kyma-integration: "true"
+        preset-kyma-cli-stable: "true"
+        preset-kyma-eventing-auth-ias-creds: "true"
```
Comment (Ressetkk) [inline prow/jobs/kyma/kyma-integration-gardener-eventing.yaml:169]: Please, check if these presets are needed.

File: prow/jobs/kyma/kyma-integration-gardener-eventing.yaml [hunk 2]
```diff
@@ -148,4 +148,74 @@ presubmits: # runs on PRs
...
+      spec:
+        containers:
+          - image: "eu.gcr.io/kyma-project/test-infra/kyma-integration:v20230414-8e724501"
+            securityContext:
+              privileged: false
+              seccompProfile:
+                type: RuntimeDefault
+              allowPrivilegeEscalation: false
+            command:
+              - "/home/prow/go/src/github.com/kyma-project/test-infra/prow/scripts/cluster-integration/kyma-integration-gardener-eventing-with-auth-manager.sh"
```
Comment (Ressetkk) [inline prow/jobs/kyma/kyma-integration-gardener-eventing.yaml:197]: We stopped allowing adding new copy-pasted scripts. Please consider moving it away to your target repository, or write your own script specifically meant to reflect the actions you need to do. That also includes creating gardener cluster and installing kyma.

File: prow/jobs/kyma/kyma-integration-gardener-eventing.yaml [hunk 3]
```diff
@@ -148,4 +148,74 @@ presubmits: # runs on PRs
...
+      cluster: untrusted-workload
+      max_concurrency: 10
+      branches:
+        - ^master$
+        - ^main$
+      extra_refs:
+        - org: kyma-project
+          repo: test-infra
+          path_alias: github.com/kyma-project/test-infra
+          base_ref: main
```
Comment (Ressetkk) [inline prow/jobs/kyma/kyma-integration-gardener-eventing.yaml:187]: If you remove the script from test-infra you will be able to remove extra_refs pointing to this repo.

File: prow/jobs/kyma/kyma-integration-gardener-eventing.yaml [hunk 4]
```diff
@@ -148,4 +148,74 @@ presubmits: # runs on PRs
...
+        preset-kyma-cli-stable: "true"
+        preset-kyma-eventing-auth-ias-creds: "true"
+      run_if_changed: '^((tests/fast-integration\S+|resources/eventing\S+)(\.[^.][^.][^.]+$|\.[^.][^dD]$|\.[^mM][^.]$|\.[^.]$|/[^.]+$))'
+      optional: true
+      skip_report: false
+      decorate: true
+      decoration_config:
+        grace_period: 600000000000
+        timeout: 14400000000000
+      path_alias: github.com/kyma-project/kyma
```
Comment (Ressetkk) [inline prow/jobs/kyma/kyma-integration-gardener-eventing.yaml:177]: You don't need this field. It's already not needed by Prow.

**Review by Ressetkk — CHANGES_REQUESTED** [review]
The script needs to go. I cannot accept PR with a new script, especially since we work to remove the dependency to test-infra from ProwJobs.

**Comment by marcobebway** [comment]
> The script needs to go. I cannot accept PR with a new script, especially since we work to remove the dependency to test-infra from ProwJobs.

As agreed, reused the same script in both prow jobs.

**Review by Ressetkk — APPROVED** [review]

**Pull request merged**